forte/train_preprocessor.py

```diff
@@ -0,0 +1,355 @@
+
+        dataset = DataPackDataset(data_source,
+                                  schemes,
+                                  self._config.dataset,
+                                  self.device)
+        iterator = DataIterator(dataset)
+
+        return iterator
+
+    @property
+    def request(self) -> Dict:
+        # pylint: disable=line-too-long
+        r"""A `Dict` containing all the information needed for doing the
+        pre-processing. This is obtained via parsing the input `request`
+
+        An example `request` is:
+
+        .. code-block:: python
+
+            request = {
+                "scope": ft.onto.Sentence
+                "schemes": {
+                    "text_tag": {
+                        "extractor": forte.data.extractor.AttributeExtractor,
+                        "converter": forte.data.converter.Converter,
+                        "type": TrainPreprocessor.DATA_INPUT,
+                    },
+                    "char_tag" {
+                        "extractor": forte.data.extractor.CharExtractor,
+                        "converter": forte.data.converter.Converter,
+                        "type": TrainPreprocessor.DATA_INPUT,
+                    }
+                    "ner_tag": {
+                        "extractor":
+                            forte.data.extractor.BioSeqTaggingExtractor,
+                        "converter": forte.data.converter.Converter,
+                        "type": TrainPreprocessor.DATA_OUTPUT,
+                    }
+                }
+            }
+
+    Here:
+
+        `"scope"`: Entry
+            A class of type :class:`forte.data.ontology.core.Entry` The
+            granularity to separate data into different examples. For example,
+            if `scope` is :class:`ft.onto.base_ontology.Sentence`, then each
+            training example will represent the information of a sentence.
+
+        `"schemes"`: `Dict`
+            A `Dict` containing the information about doing the pre-processing.
+            The `key` is the tags provided by input `request`. The `value` is a
+            `Dict` containing the information for doing pre-processing for that
+            feature.
+
+        `"schemes.tag.extractor"`: Extractor
+            An instance of type :class:`forte.data.extractor.BaseExtractor`.
+
+        `"schemes.tag.converter"`: Converter
+            An instance of type :class:`forte.data.converter.Converter`.
+
+        `"schemes.tag.type"`: TrainPreprocessor.DATA_INPUT/DATA_OUTPUT
+            Denoting whether this feature is the input or output feature.
+        """
+        if not self._request:
+            self._parse_request(self._request)
+        return self._request
+
+    @property
+    def user_request(self) -> Dict:
+        r"""A `Dict` passed by users when
+        :class:`forte.train_preprocessor.TrainPreprocessor` instance is created.
+        Please refer to class documentation for the detailed format.
```

Comment:
The class documentation does not contain the detailed format.

Reply:
Sorry I think this property is deprecated. We only have the property `request`. I will remove this.